fastq_filtrator.py

```diff
@@ -0,0 +1,96 @@
+
+def fun_gc_bounds(gc_bounds, GC_content):
+    left_border = 0.0
+    rigth_border = 100.0
+    if isinstance(gc_bounds, float) | isinstance(gc_bounds, int) == True:
+        rigth_border = float(gc_bounds)
+    else:
+        left_border = float(gc_bounds[0])
+        rigth_border = float(gc_bounds[1])
+    if left_border < GC_content < rigth_border:
+        return(True)
+    else:
+        return(False)
+
+
+def fun_length_bounds(length_bounds, counter):
+    left_border = 0.0
+    rigth_border = float(2**32)
+    if isinstance(length_bounds, float) | isinstance(length_bounds, int) == True:
+        rigth_border = float(length_bounds)
+    else:
+        left_border = float(length_bounds[0])
+        rigth_border = float(length_bounds[1])
+    if left_border < counter < rigth_border:
+        return(True)
+    else:
+        return(False)
+
+
+def fun_qual_thresh(qual_strok, quality_threshold = 0.0):
+    qual = 0
+    cou = 0
+    for k in qual_strok:
+        qual += ord(k)
+        cou += 1
+    if (qual/cou) < quality_threshold:
+        return(False)
+    else:
+        return(True)
+
+
+
+passed = "_passed.fastq"
+
+
+def main(input_fastq, output_file_prefix, gc_bounds = (0, 100), length_bounds = (0, 2**32), quality_threshold = 0, save_filtered = False):
+    with open(input_fastq, "r") as file, open(f"{output_file_prefix}{passed}", "w") as ouf:
+        spis = []
+        for line in file:
+            spis += [line.strip()]

```

Comment:
Лучше стараться не считывать файл целиком в память, а обрабатывать его построчно по ходу прочтения. Но даже если принять этот вариант, то его можно записать проще
```suggestion
        spis = file.readlines()
```